src/LinearSolverTypes.f90

```diff
@@ -1073,6 +1127,9 @@ SUBROUTINE solve_LinearSolverType_Iterative(solver)
       ENDSELECT
 
     CASE(GMRES)
+      ! Ensure setupPrecond has been called
+      IF (.NOT. ALLOCATED(solver%preCondType) &
+          .AND. solver%pcTypeName /= "NOPC") CALL solver%setupPC()
```

Comment:
Would it be useful to add a logical to the base or specific linear solver type for `%hasSetPC` or something like that?  Then you can just check that value instead?  Or maybe on the preconditioner type?

Reply:
Yeah, I debating doing that. I'll add it and check that instead.